helios-testing/src/main/java/com/spotify/helios/testing/SoloHostProber.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.testing;
+
+import com.google.common.net.HostAndPort;
+
+import com.spotify.helios.client.HeliosClient;
+
+import java.util.concurrent.Callable;
+
+/**
+ * A interface that extends {@link Callable}.
+ * This interface is used by {@link HeliosSoloDeployment} to check whether there are agents
+ * available in helios-solo. The main purpose of this interface is to make HeliosSoloDeployment
+ * easy to test by allowing us to easily mock the helios client calls to the master in helios-solo.
+ */
+interface SoloHostProber extends Callable<Boolean> {
+
+  void setHostAndPort(HostAndPort hostAndPort);
+
+  void setHeliosClient(HeliosClient heliosClient);
```

Comment:
Assuming that the purpose of extending `Callable` is to use it with `Polling.awaitUnchecked(..)`, I think it would be clearer instead to have something like

```java
interface SoloHostProber {
    boolean check(HeliosClient client, HostAndPort hap);
}

// in the place where it is used
Polling.awaitUnchecked(time, new Callable() { return soloHostProber.check(client, hap); });
```

then you don't have to remember to properly call some methods in-order before others. It would also reduce boilerplate in the implementations of these interfaces.